rust/cymbal/src/symbol_store/caching.rs

```diff
@@ -151,6 +134,27 @@ impl CacheInner {
 }
 
 struct CachedSymbolSet {
-    pub data: Arc<Vec<u8>>,
+    pub data: Arc<dyn Any + Send + Sync>,
+    pub bytes: usize,
     pub last_used: Instant,
 }
+
+pub trait Cacheable: Any + Send + Sync {
+    fn bytes(&self) -> usize;
+}
+
+impl Cacheable for Vec<u8> {
+    fn bytes(&self) -> usize {
+        self.len()
+    }
+}
+
+impl Cacheable for SourceMap {
+    fn bytes(&self) -> usize {
+        // This is an extremely expensive way to get the size of a sourcemap, but we're more-or-less ok with that,
+        // since we only call it when we add an item to the cache, which should be rare.
+        let mut data = vec![];
+        self.to_writer(&mut data).unwrap();
```

Comment:
Assuming it's ok to add the `unwrap` here?

Reply:
Yeah it's fine, it only fails if the underlying writer fails I think, and a vec can never fail a write (or, a vec failing a write is an oom, not an `Err`).